syntax = "proto3";

package order_proto;


// The order service definition.
service Order {
  // Add an order
  rpc AddOrder (OrderInformation) returns (addOrderResult) 
  {
  }

  // query an Order
  rpc QueryOrder (OrderID) returns (OrderInformation) 
  {
  }
}

message OrderInformation {
  int64 merchant_id = 1;
  string merchant_order_id = 2;
  string description = 3;
  string address = 4;
  double amount = 5;
  string currency = 6;
}

message addOrderResult {
  bool result = 1;
}

message OrderID {
  int64 merchant_id= 1;
  string merchant_order_id= 2;
}